src/gui/qgsmapcanvas.cpp

```diff
@@ -2531,25 +2529,6 @@ void QgsMapCanvas::mouseReleaseEvent( QMouseEvent *e )
     // call handler of current map tool
     if ( mMapTool )
     {
-      // right button was pressed in zoom tool? return to previous non zoom tool
-      if ( e->button() == Qt::RightButton && mMapTool->flags() & QgsMapTool::Transient )
```

Comment:
If this code block is removed, the flag is not actually "deprecated" but already disabled. Am I understanding something wrong?

Reply:
I think that's ok. We've done it before where "optional" functionality like this is soft broken and no longer has an effect when it's no longer desirable.  The important thing is avoiding the exceptions.